src/ideal/ideal.jl

```diff
@@ -1278,6 +1278,36 @@ function hilbert_series(I::sideal{spoly{T}}, w::Vector{<:Integer}) where T <: Ne
    return z
 end
 
+@doc raw"""
+    hilbert_series(I::sideal{spoly{T}}, Qt::PolyRing) where T <: Nemo.FieldElem
+
+Return the polynomial $Q(t)$ as element of Qt where `Q(t)/(1-t)^nvars(base_ring(I))`
+is the Hilbert-Poincare series of $I$ for weights $(1, \dots, 1)$.
+"""
+function hilbert_series(I::sideal{spoly{T}}, Qt::PolyRing) where T <: Nemo.FieldElem
+   I.isGB || error("Not a Groebner basis")
+   R = base_ring(I)
+   GC.@preserve I R new_ptr = libSingular.scHilbPoly(I.ptr, R.ptr, Qt.ptr)
+   return Qt(new_ptr)
+end
+
+@doc raw"""
+    hilbert_series(I::sideal{spoly{T}}, w::Vector{<:Integer}) where T <: Nemo.FieldElem
+
+Return the polynomial $Q(t)$ of Qt where $\frac{Q(t)}{\prod_i (1-t^{w_i})}$
+is the Hilbert-Poincare series of $I$ for weights $\{w_i\}$. Each weight must be
+positive $w_i > 0$.
+"""
+function hilbert_series(I::sideal{spoly{T}}, w::Vector{<:Integer}, Qt::PolyRing) where T <: Nemo.FieldElem
+   I.isGB || error("Not a Groebner basis")
+   R = base_ring(I)
+   length(w) == nvars(R) || error("wrong number of weights")
```

Comment:
Is it useful in the error mesg to say how many weights were received, and how many were expected?